src/components/Header.tsx

```diff
@@ -0,0 +1,111 @@
+import classNames from 'classnames';
+import { ErrorMessage } from '../types/ErrorMessage';
+import { Todo } from '../types/Todo';
+import { addTodo, changeTodo, USER_ID } from '../api/todos';
+import { useEffect, useState } from 'react';
+
+type Props = {
+  todos: Todo[];
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setErrorMessage: (e: ErrorMessage) => void;
+  title: string;
+  setTitle: (t: string) => void;
+  setTempTodo: (t: Todo | null) => void;
+  inputRef: React.RefObject<HTMLInputElement>;
+};
+export const Header = ({
+  todos,
+  setTodos,
+  setTempTodo,
+  title,
+  setTitle,
+  setErrorMessage,
+  inputRef,
+}: Props) => {
+  const [formLoading, setFormLoading] = useState(false);
+
+  useEffect(() => {
+    if (!formLoading && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [formLoading, inputRef]);
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
```

Comment:
It is important to follow the same way of creating functions everywhere
```suggestion
  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
```